arcconfig/arcconfig.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package arcconfig defines the API Runtime Control config and provides
+// the config loading and validation functions.
+package arcconfig
+
+import "time"
+
+// APIRuntimeControlConfig is an Apigee Environment-level config for
+// Envoy Adapter. It contains a list of operations for the adapter to
+// perform request authentication and authorization.
+type APIRuntimeControlConfig struct {
+	// Name of the config
+	Name string `yaml:"name"`
+
+	// Revision of the config
+	Revision string `yaml:"revision"`
```

Comment:
Are we using the name or revision or both for matching? We need to document what these mean wrt. our logic.

Reply:
As replied in other comments. Will use ID only and require its uniqueness.